File: src/Server/Infrastructure/Persistence/Entities/AuditLog.cs
using Server.Core.Common;
using Server.Infrastructure.Identity;
using Server.Infrastructure.Persistence.Entities.Enums;

namespace Server.Infrastructure.Persistence.Entities;

public class AuditLog : BaseEntity
{
    public Guid? UserId { get; set; }

    public string? UserName { get; set; }

    public AuditAction Action { get; set; }

    public string EntityType { get; set; } = string.Empty;

    public Guid? EntityId { get; set; }

    public string? EntityDisplayName { get; set; }

    public string? Changes { get; set; }

    public string? IpAddress { get; set; }

    public string? UserAgent { get; set; }

    public DateTime Timestamp { get; set; } = DateTime.UtcNow;

    public string? AdditionalInfo { get; set; }

    public ApplicationUser? User { get; set; }
}

File: src/client/src/types/audit-log.ts
export enum AuditAction {
  Create = 0,
  Update = 1,
  Delete = 2,
  Login = 3,
  Logout = 4,
  FailedLogin = 5,
  StatusChange = 6,
  Payment = 7,
  Refund = 8,
  Print = 9,
  Export = 10,
}

export interface AuditLogChangeItem {
  field: string;
  old?: unknown;
  new?: unknown;
}

export interface AuditLogDetail {
  id: string;
  userId?: string | null;
  userName?: string | null;
  action: AuditAction;
  entityType: string;
  entityId?: string | null;
  entityDisplayName?: string | null;
  changes?: string | null;
  parsedChanges?: AuditLogChangeItem[] | null;
  ipAddress?: string | null;
  userAgent?: string | null;
  timestamp: string;
  additionalInfo?: string | null;
}

export interface AuditLogListItem {
  id: string;
  userName?: string | null;
  action: AuditAction;
  entityType: string;
  entityDisplayName?: string | null;
  timestamp: string;
  ipAddress?: string | null;
}
